exonum-java-binding/core/src/test/java/com/exonum/binding/core/storage/indices/ProofMapIndexProxyIntegrationTest.java

```diff
@@ -663,6 +687,7 @@ void clearFailsIfSnapshot() {
    * - ProofMap constructor preserves the index type and
    * - Map constructor checks it, preventing illegal access to ProofMap internals.
    */
+  @Disabled // FIXME: Panics on failed assertion "Index type doesn't match specified".
```

Comment:
Doesn't the panic include the error details? Shan't the string in the assertion be updated? If the message isn't good enough, it must be updated in the core.

Reply:
It looks like the assertion message at the core side has been changed to just "Index type doesn't match specified". If I change the assertion string test passes. Are we happy with that message or should I request a more detailed message from the core team?

Reply:
The proper error message must include the index type in the database and (here — optionally) the attempted type.

Reply:
[ECR-3354](https://jira.bf.local/browse/ECR-3354)